server/src/projects/project-applicants/project-applicant.controller.ts

```diff
@@ -0,0 +1,85 @@
+import { Controller, Patch, Body, Param, Post, UseInterceptors, UseGuards, UsePipes, Delete, HttpException, HttpStatus } from '@nestjs/common';
+import { pick } from 'underscore';
+import { CrmService } from '../../crm/crm.service';
+import { JsonApiSerializeInterceptor } from '../../json-api-serialize.interceptor';
+import { AuthenticateGuard } from '../../authenticate.guard';
+import { JsonApiDeserializePipe } from '../../json-api-deserialize.pipe';
+import { PROJECTAPPLICANT_ATTRS } from './project-applicants.attrs';
+import { CONTACT_ATTRS } from '../../contact/contacts.attrs';
+
+const ACTIVE_CODE = 1;
+
+@UseInterceptors(new JsonApiSerializeInterceptor('project-applicants', {
+  id: 'dcp_projectapplicantid',
+  attributes: [
+    ...PROJECTAPPLICANT_ATTRS,
+
+    'contact',
+  ],
+
+  contact: {
+    ref: 'contactid',
+    attributes: CONTACT_ATTRS,
+  },
+}))
+@UseGuards(AuthenticateGuard)
+@UsePipes(JsonApiDeserializePipe)
+@Controller('project-applicants')
+export class ProjectApplicantController {
+  constructor(private readonly crmService: CrmService) {}
+
+  @Post('/')
+  async create(@Body() body) {
+    const allowedAttrs = pick(body, PROJECTAPPLICANT_ATTRS);
+    const email = body.emailaddress;
+    const projectId = body.project;
+    const fullName = body.dcp_name;
+
+    let contactId;
```

Comment:
can you move contactId down to maybe line 47? It feels a bit far from where it's actually used